integration/app/store/todos/todos.state.ts

```diff
@@ -22,9 +24,15 @@ export class TodosState {
 
   @Action(SetPrefix)
   public setPrefix({ getState, setState, patchState }: StateContext<TodoStateModel>) {
-    const state: TodoStateModel = getState();
-    const pizza1 = state.pizza.model.toppings;
-    patchState({ pizza: { model: { toppings: 'Mr. ' + pizza1 } } });
+    setState(
+      patch({
+        pizza: patch({
+          model: patch({
+            toppings: (topping: any) => 'Mr. ' + topping
+          })
+        })
+      })
+    );
```

Comment:
how well does the type inference work when combining patches like this?

Reply:
I think bad, but maybe we can usage generic for patch<T>

Reply:
@deebloo It works very well. It took me 3 hours to get the operator done with tests and about 12 frustrating hours to get the type checking working nicely. Have a look at the bigger examples in the tests. 
You can supply a type T here for the patch operator. Unfortunately this integration example lacked decent types and is deeply nested, not the ideal use case for the patch operator.

I agree that operators like those suggested by @splincode would be better suited for this particular scenario.